src/shims/unix/sync.rs

```diff
@@ -898,6 +885,39 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
     }
 }
 
+struct Callback<'tcx> {
```

Comment:
This is also just used in one place so could be moved inside `pthread_cond_timedwait` -- or else it should get a descriptive name + docs.